pyhoma/models.py

```diff
@@ -277,3 +389,245 @@ class DataType(Enum):
+    DBUS_SPAWN_FAILED = 231
+    DBUS_SPAWN_SETUP_FAILED = 232
+    DBUS_SPAWN_CONFIG_INVALID = 233
+    DBUS_SPAWN_SERVICE_INVALID = 234
+    DBUS_SPAWN_SERVICE_NOT_FOUND = 235
+    DBUS_SPAWN_PERMISSIONS_INVALID = 236
+    DBUS_SPAWN_FILE_INVALID = 237
+    DBUS_SPAWN_NO_MEMORY = 238
+    DBUS_UNIX_PROCESS_ID_UNKNOWN = 239
+    DBUS_INVALID_SIGNATURE = 240
+    DBUS_INVALID_FILE_CONTENT = 241
+    DBUS_SELINUX_SECURITY_CONTEXT_UNKNOWN = 242
+    DBUS_ADT_AUDIT_DATA_UNKNOWN = 243
+    DBUS_OBJECT_PATH_IN_USE = 244
+    DBUS_INCONSISTENT_MESSAGE = 245
+    NOT_IMPLEMENTED_YET = 300
+    MODULE_NOT_LOADED = 301
+    APPLICATION_NOT_RUNNING = 302
+    NONEXEC_MANUALLY_CONTROLLED = 400
+    NONEXEC_AUTOMATIC_CYCLE = 401
+    NONEXEC_BATTERY_LEVEL = 402
+    NONEXEC_WRONG_LOAD_CONNECTED = 403
+    NONEXEC_HIGH_CONSUMPTION = 404
+    NONEXEC_LOW_CONSUMPTION = 405
+    NONEXEC_COLOUR_NOT_REACHABLE = 406
+    NONEXEC_USER_ACTION_NEEDED = 407
+    NONEXEC_COMMAND_INCOMPATIBLE_WITH_MOVEMENT = 408
+    NONEXEC_CANNOT_CHANGE_STATE = 409
+    NONEXEC_FILTER_MAINTENANCE = 410
+    NONEXEC_OPERATING_MODE_NOT_SUPPORTED = 411
+    WHILEEXEC_MANUALLY_CONTROLLED = 420
+    WHILEEXEC_AUTOMATIC_CYCLE = 421
+    WHILEEXEC_BATTERY_LEVEL = 422
+    WHILEEXEC_WRONG_LOAD_CONNECTED = 423
+    WHILEEXEC_HIGH_CONSUMPTION = 424
+    WHILEEXEC_LOW_CONSUMPTION = 425
+    WHILEEXEC_COLOUR_NOT_REACHABLE = 426
+    WHILEEXEC_USER_ACTION_NEEDED = 427
+    WHILEEXEC_COMMAND_INCOMPATIBLE_WITH_MOVEMENT = 428
+    WHILEEXEC_CANNOT_CHANGE_STATE = 429
+    WHILEEXEC_FILTER_MAINTENANCE = 430
+    WHILEEXEC_OPERATING_MODE_NOT_SUPPORTED = 431
+    OVERRIDEMODE_ERROR = 450
+    CAMERA_INVALID_CREDENTIALS = 500
+    UNSUPPORTED_CAMERA_TYPE = 501
+    NETWORK_COULDNT_RESOLVE_HOST = 601
+    NETWORK_COULDNT_CONNECT = 602
+    NETWORK_OPERATION_TIMEDOUT = 603
+    LPB_APP_OUT_OF_RANGE = 701
+    LPB_APP_OUT_OF_MAXRANGE = 702
+    LPB_APP_OUT_OF_MINRANGE = 703
+    LPB_APP_MEMORY_ERROR = 704
+    LPB_APP_READ_ONLY = 705
+    LPB_APP_ILLEGAL_CMD = 706
+    LPB_APP_VOID_DP = 707
+    LPB_APP_TYPE_CONFLICT = 708
+    LPB_APP_READ_CMD_INCORRECT = 709
+    LPB_APP_WRITE_CMD_INCORRECT = 710
+    LPB_APP_CMD_TYPE_INCORRECT = 711
+    LPB_APP_WRITE_TIMEOUT = 712
+    LPB_APP_CANNOT_WRITE_GW = 713
+    LPB_APP_UNKNOWN_GATEWAY = 714
+    LPB_APP_GATEWAY_UNREACHABLE = 715
+    APPLICATION_ERROR = 800
+    HUE_INVALID_CREDENTIALS = 900
+    HUE_LINK_BUTTON_NOT_PRESSED = 901
+    HUE_DEVICE_IS_OFF = 902
+    TIMED_OUT = 10001
+    CANCELLED = 10002
+    UNKNOWN_ERROR_CODE = 10003
+    SERVER_FAILURE = 10004
+    PEER_DOWN = 10005
+    GATEWAY_BUFFER_OVERFLOW = 10006
+    UNKNOWN_DETAILED_ERROR = 10007
+
+
+class UpdateBoxStatus(Enum):
+    NOT_UPDATABLE = "NOT_UPDATABLE"
+    READY_TO_UPDATE = "READY_TO_UPDATE"
+    READY_TO_BE_UPDATED_BY_SERVER = "READY_TO_BE_UPDATED_BY_SERVER"
+    READY_TO_UPDATE_LOCALLY = "READY_TO_UPDATE_LOCALLY"
+    UP_TO_DATE = "UP_TO_DATE"
+    UNKNOWN = "UNKNOWN"
```

Comment:
It's nice to have these enums. But where should we use them?

Reply:
These enums are used by `Gateway` and could be used to check if there is an update available. Some of the other enums are not for Gateway and need to be mapped in future PR's. I just wanted to include them already before I lose the source again :D.

Reply:
@tetienne included a few already. Would make our coordinator code a bit more strongly typed using those enums.

Reply:
That's great!

Reply:
This class is now really huge. What do you think about moving all the enums to a dedicated module?

Reply:
> This class is now really huge. What do you think about moving all the enums to a dedicated module?

Should I create an `enums.py`?